vscode/microsoft-kiota/src/extension.ts

```diff
@@ -0,0 +1,248 @@
+          );
+          return;
+        }
+        if (
+          !vscode.workspace.workspaceFolders ||
+          vscode.workspace.workspaceFolders.length === 0
+        ) {
+          vscode.window.showErrorMessage(
+            "No workspace folder found, open a folder first"
+          );
+          return;
+        }
+        const config = await generateSteps();
+        if (!openApiTreeProvider.descriptionUrl) {
+          vscode.window.showErrorMessage(
+            "No description url found, select a description first"
+          );
+          return;
+        }
+        await generateClient(
+          openApiTreeProvider.descriptionUrl,
+          typeof config.outputPath === "string" ? config.outputPath : './output',
+          typeof config.language === "string" ? parseGenerationLanguage(config.language) : KiotaGenerationLanguage.CSharp,
+          selectedPaths,
+          [],
+          typeof config.clientClassName === "string" ? config.clientClassName : 'ApiClient',
+          typeof config.clientNamespaceName === "string" ? config.clientNamespaceName : 'ApiSdk');
+    }),
+    vscode.commands.registerCommand(
+      `${extensionId}.searchApiDescription`,
+      async () => {
+        const config = await searchSteps(searchDescription);
+        if(config.descriptionPath) {
+          openApiTreeProvider.descriptionUrl = config.descriptionPath;
+          vscode.commands.executeCommand(`${treeViewId}.focus`);
+        }
+    })
+  );
+
+  // create a new status bar item that we can now manage
+  kiotaStatusBarItem = vscode.window.createStatusBarItem(
+    vscode.StatusBarAlignment.Right,
+    100
+  );
+  kiotaStatusBarItem.command = statusBarCommandId;
+  context.subscriptions.push(kiotaStatusBarItem);
+
+  // update status bar item once at start
+  await updateStatusBarItem();
+  let disposable = vscode.commands.registerCommand(
+    "microsoft-kiota.updateClients",
+    async () => {
+      if (
+        !vscode.workspace.workspaceFolders ||
+        vscode.workspace.workspaceFolders.length === 0
+      ) {
+        vscode.window.showErrorMessage(
+          "No workspace folder found, open a folder first"
+        );
+        return;
+      }
+      await updateStatusBarItem();
+      try {
+        kiotaOutputChannel.clear();
+        kiotaOutputChannel.show();
+        kiotaOutputChannel.info(
+          `updating workspace with path ${vscode.workspace.workspaceFolders[0].uri.fsPath}`
+        );
+        await connectToKiota(async (connection) => {
+          const request = new rpc.RequestType<string, KiotaLogEntry[], void>(
+            "Update"
+          );
+          const result = await connection.sendRequest(
+            request,
+            vscode.workspace.workspaceFolders![0].uri.fsPath
+          );
+          const informationMessages = result.filter((x) => x.level === 2);
+          const errorMessages = result.filter(
+            (x) => x.level === 5 || x.level === 4
```

Comment:
We could declare an enum with the levels to make the levels more intuitive clearer

Reply:
added, pushing later with another set of code changes